### DISCUSSION TITLE
Add 'between' operator to the addNumericRefinement method

### DISCUSSION THREAD

**Discussion by seafoox** [algolia/instantsearch#5745] [Ideas]
When using the API clients it's possible to pass a range of numeric filters.
For instance if I want to retrieve records with a price within 10 and 100, I just need to do.
```
index.search('', {
  filters: 'price:10 TO 100'
}
```
It would be nice to be able to do the same using the helpers' `addNumericRefinement` method.
I was thinking that maybe it could be used that way

## Basic example
```
helper.addNumericRefinement('price', 'to', '10-100')
```

## Advanced example
```
helper.addNumericRefinement('code', '=', '1');
helper.addNumericRefinement('price', 'to', ['1000-3000', '10-100']);
helper.setQuery('los');
helper.search();
```
which would be the same thing as doing 
```
index.search('los', {
  filters: 'code=1 AND (price:1000 TO 3000 OR price:10 TO 100)'
}
```
when using the JS API client

**Comment by ehayman** [comment]
A much belated +1 -> currently feeling the pain of not having this in the helper :) Would be an awesome addition

**Comment by Haroenv** [comment]
```js
helper.addNumericRefinement('price', 'to', [10, 100]);
helper.addNumericRefinement('price', 'to', [1000, 3000]);
```

This would seem like a more logical API I think. Seems to make sense and a few people have upvoted the issue. Not currently on our plan, but possible

**Comment by ehayman** [comment]
@Haroenv agreed, that would make sense to me.

To help sway you, right now building out a common UX pattern for price ranges (like the mockup) is quite _quite_ painful:

![image](https://user-images.githubusercontent.com/8277212/42703434-771f8d3a-8681-11e8-8ef0-f6b395cdf7d5.png)

This is a pretty common ask I would argue we should support more easily.

**Comment by Haroenv** [comment]
Isn’t this quite easily possible with the PriceRanges component (IS.js, but if a valid usecase, should be brought to the others)?
